content/docs/internals/architecture.md

```diff
@@ -11,38 +11,46 @@ Pomerium sits between end users and services requiring strong authentication. Af
+1. The Proxy service receives the request and redirects it to the Authorization service before proxying it
+1. If the Authorization service doesn’t see a session cookie, the Proxy service redirects the request to the Authentication service to verify the client’s identity
+1. After the Authentication service verifies the client’s identity and saves a local session cookie, the Proxy service redirects the session data to the Databroker service
+1. Before redirecting the client to the target resource, the Proxy service checks permissions with the Authorization service, then maps a route based on the internal and external routes defined in the route’s policy
 
-  - Handles authentication flow to your IdP as needed
-  - Handles identity verification after initial Authentication
-  - Establishes user session cookie
-  - Stores user OIDC tokens in Proxy service
+### Authentication service
+The Authentication service is responsible for authenticating users against an Identity Provider (IdP) and establishing user sessions. By incorporating OAuth 2.0 and OIDC protocols into the authentication flow, the Authentication service provides single sign-on authentication that enables it to extract user identity details and session data necessary for managing Pomerium sessions. 
 
-- Authorization Service
+At a high level, when the Authentication service first receives a request from the Proxy service, it:
+1. Sends an HTTP request on behalf of the client to the IdP
+1. Completes the authentication flow with the IdP and extracts relevant session data, such as device credentials, user ID, issuance and expiration times, OAuth tokens, and any OIDC claims (scopes) provided by the IdP
+1. Saves session data to a local session cookie and redirects the client with session data encrypted in URL parameters to the Proxy service
+1. Verifies user identity after the session expires
 
-  - Processes policy to determine permissions for each service
-  - Handles authorization check for all user sessions
-  - Directs Proxy service to initiate Authentication flow as required
-  - Provides additional security related headers for upstream services to consume
+### Authorization service
+The Authorization service processes policies to determine what permissions the client has. Each request sent by the client must first go through an authorization check before the Proxy service proxies the request. 
 
-- Databroker Service
+When the Authorization service receives a request from the Proxy service, the following actions take place:
+1. The Authorization service first looks for a session cookie, which contains the client’s JWT. If no session cookie is present, the Authorization service sends a request to the Authentication service to establish session identity. 
+1. Once a session cookie is in place, the Proxy service redirects the request to the Authorization service so it can determine permissions based on JWT claims and policy. 
```

Comment:
The Proxy service does not redirect the request to the Authorization service. It makes a gRPC call.